src/panel/applets/status/PowerIndicator.vala

```diff
@@ -324,12 +311,9 @@ public class PowerIndicator : Gtk.Bin {
 	}
 
 	void has_power_profiles() {
-		if (profiles_proxy == null) {
-			Bus.get_proxy.begin<PowerProfilesDBus>(BusType.SYSTEM, POWER_PROFILES_DBUS_NAME, POWER_PROFILES_DBUS_OBJECT_NAME, 0, null, on_proxy_get);
-		} else {
-			// already have a proxy object
-			create_power_profiles_options();
-		}
+		if (profiles_proxy != null) create_power_profiles_options();
```

Comment:
You're missing a return here. With the code now, it will always call to get the proxy even when it is unnecessary